components/features/Dashboard/Applications/InteractiveQuiz/Assignment/Assignment.tsx

```diff
@@ -1,45 +1,54 @@
-import usePostRequest from "@hooks/shared/usePostRequest";
-import { useRouter } from "next/router";
 import React from "react";
 import AssignmentTimer from "../AssignmentTimer";
 import styles from "./Assignment.module.css";
 import FileDropzone from "./FileDropzone";
 
 interface Props {
   assignment: AssignmentObject;
-  setCurrentActiveAssignment: React.Dispatch<React.SetStateAction<AssignmentObject | null>>;
+  setCurrentActiveAssignment: React.Dispatch<
+    React.SetStateAction<AssignmentObject | null>
+  >;
 }
 
 const getSecondsRemaining = (time: string) => {
-  const today = new Date();
-  const todayStr = today.toISOString();
-  const endIdx = todayStr.indexOf("T");
-  const todayDate = todayStr.substring(0, endIdx + 1);
-  const endWorkingTimeStr = todayDate.concat(time);
-  const nowWorkingTimeStr = todayDate.concat(today.toLocaleTimeString());
-  const endDate = new Date(endWorkingTimeStr);
-  const startDate = new Date(nowWorkingTimeStr);
-  const diffMs = endDate.getTime() - startDate.getTime();
-  const differenceInSeconds = Math.floor(diffMs / 1000);
-  return differenceInSeconds;
+  const timeinList = time.split(":").map((x) => parseInt(x));
+  const [hours, minutes] = timeinList;
+  let seconds: number = 0;
+  if (timeinList.length > 2) seconds = timeinList[2];
+  const date = new Date();
```

Comment:
Hello, Rashad. If I may clarify, so the logic is to convert the time sent from the backend to an ISO datestring by using today's date?

Reply:
Yes Johanes, that's correct. At the moment we have to convert it back to a ISO string and derive the time remaining from there.

Reply:
Is there anything that I could do in the backend to reduce the logic in the front end?

Reply:
I think it would be better if the ISO is directly sent from the backend. Since, at the moment, we have no way to tell when the assignment should end, if it is still somehow accessible in the following day.

Reply:
Noted, Shad. I will implement that in this week.